src/mail.sh

```diff
@@ -77,18 +77,65 @@ function mail_main()
 function mail_send()
 {
   local flag="$1"
+  local opts="${configurations[send_opts]}"
   local recipients="${options_values['TO']}"
+  local rev_list="${options_values['REV_LIST']}"
   local extra_opts="${options_values['PASSTHROUGH']}"
+  local kernel_root
   local cmd='git send-email'
 
   flag=${flag:-'SILENT'}
 
-  [[ -n "$recipients" ]] && cmd+=" --to=$recipients"
+  [[ -n "$recipients" ]] && recipients="--to=$recipients"
+
+  kernel_root="$(find_kernel_root "$PWD")"
+  # if inside a kernel repo use get_maintainer to populate recipients
+  if [[ -n "$kernel_root" ]]; then
+    recipients+=" $(generate_kernel_recipients "$kernel_root" "$rev_list")"
+  fi
+
+  [[ -n "$opts" ]] && cmd+=" $opts"
+  [[ -n "$recipients" ]] && cmd+=" $recipients"
   [[ -n "$extra_opts" ]] && cmd+=" $extra_opts"
 
   cmd_manager "$flag" "$cmd"
 }
 
+function generate_kernel_recipients()
+{
+  local kernel_root="$1"
+  local revision="$2"
+  local -a to_list=()
+  local to=''
+  local -a cc_list=()
+  local cc=''
+  local out=''
+  local patch_cache="${KW_CACHE_DIR}/patches/"
+  local get_maintainer="perl $kernel_root/scripts/get_maintainer.pl --nogit --nogit-fallback --no-r --no-n --multiline --nokeywords --norolestats"
+
```

Comment:
Add a note in the documentation that developers should ensure that we are in the kernel tree before calling this function.